staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/packages.go

```diff
@@ -81,6 +82,12 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 
 		var toGenerate []applyConfig
 		for _, t := range p.Types {
+			// If we don't have an ObjectMeta field, we lack the information required to make the Apply or ApplyStatus call
+			// to the kube-apiserver, so we don't need to generate the type at all
+			clientTags := genclientTags(t)
+			if clientTags.GenerateClient && !hasObjectMetaField(t) {
+				continue
```

Comment:
optional: maybe log a trace message that we're skipping this, to save future selves trouble trying to debug why a type isn't getting generated?